src/cpr_data_access/utils.py

```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
         elif not obj:
             return default
     return obj
+
+
+def unflatten_json(data: dict) -> dict:
+    """
+    Unflatten a dictionary with keys that are dot-separated strings.
+
+    I.e. metadata.data respresents {"metadata": {"data": {}}}
+    """
+    unflattened = {}
+    for key, value in data.items():
+        parts = key.split(".")
+        current = unflattened
+        for part in parts[:-1]:
+            current = current.setdefault(part, {})
+        current[parts[-1]] = value
+    return unflattened
+
+
+def remove_if_all_nested_vals_none(data: dict, key: str) -> dict:
+    """Remove an item if it's a dict with all values in the nested dictionary are None."""
```

Comment:
This documentation says to me that `data[key]` can be nested, but this only works if it's flat. Maybe worth updating.

Also, tbh, given this is two lines of readable code used in one place I'd consider not using a separate function here